qt/scientific_interfaces/Indirect/test/IndirectDataTablePresenterTest.h

```diff
@@ -0,0 +1,368 @@
+    UNUSED_ARG(spectrum);
+    return "";
+  };
+
+  std::vector<std::string> getSpectrumDependentAttributes() const override {
+    return {};
+  };
+};
+
+GNU_DIAG_ON_SUGGEST_OVERRIDE
+
+class IndirectDataTablePresenterTest : public CxxTest::TestSuite {
+public:
+  /// Needed to make sure everything is initialized
+  IndirectDataTablePresenterTest() { FrameworkManager::Instance(); }
+
+  static IndirectDataTablePresenterTest *createSuite() {
+    return new IndirectDataTablePresenterTest();
+  }
+
+  static void destroySuite(IndirectDataTablePresenterTest *suite) {
+    delete suite;
+  }
+
+  void setUp() override {
+    m_model = std::make_unique<NiceMock<MockIndirectDataTableModel>>();
+    createEmptyTableWidget(5, 5);
+    m_presenter = std::make_unique<IndirectDataTablePresenter>(
+        std::move(m_model.get()), std::move(m_table.get()));
+
+    SetUpADSWithWorkspace ads("WorkspaceName", createWorkspace(5));
+    m_model->addWorkspace("WorkspaceName");
+  }
+
+  void tearDown() override {
+    AnalysisDataService::Instance().clear();
+
+    TS_ASSERT(Mock::VerifyAndClearExpectations(m_table.get()));
+    TS_ASSERT(Mock::VerifyAndClearExpectations(m_model.get()));
+
+    m_presenter.reset();
+    m_model.reset();
+    m_table.reset();
+  }
+
+  ///----------------------------------------------------------------------
+  /// Unit tests to check for successful presenter instantiation
+  ///----------------------------------------------------------------------
+
+  void test_that_the_model_has_been_instantiated_correctly() {
+    ON_CALL(*m_model, isMultiFit()).WillByDefault(Return(false));
+
+    EXPECT_CALL(*m_model, isMultiFit()).Times(1).WillOnce(Return(false));
+
+    m_model->isMultiFit();
+  }
+
+  void
+  test_that_invoking_setStartX_will_alter_the_relevant_column_in_the_table() {
+    int const startXColumn(2);
+
+    m_presenter->setStartX(2.2, 0, 0);
+
+    for (auto row = 0; row < m_table->rowCount(); ++row)
+      TS_ASSERT_EQUALS(m_table->item(row, startXColumn)->text().toStdString(),
+                       "2.2")
+  }
+
+  ///----------------------------------------------------------------------
+  /// Unit Tests that test the signals call the correct methods
+  ///----------------------------------------------------------------------
+
+  void
+  test_that_the_cellChanged_signal_will_set_the_models_startX_when_the_relevant_column_is_changed() {
+    int const startXColumn(2);
+
+    EXPECT_CALL(*m_model, setStartX(2.0, 0, 0)).Times(1);
+
+    m_table->item(0, startXColumn)->setText("2.0");
+  }
+
+  void
+  test_that_the_cellChanged_signal_will_set_the_models_endX_when_the_relevant_column_is_changed() {
+    int const endXColumn(3);
+
+    EXPECT_CALL(*m_model, setEndX(2.0, 0, 0)).Times(1);
+
+    m_table->item(0, endXColumn)->setText("2.0");
+  }
+
+  void
+  test_that_the_cellChanged_signal_will_set_the_models_excludeRegion_when_the_relevant_column_is_changed() {
+    int const excludeRegionColumn(4);
```

Comment:
You are reusing excludeRegionColumn, endXColumn etc quite often. It might be cleaner to define them as const at the top.